src/components/Inputs/TextInput/TextInput.js

```diff
@@ -53,29 +50,7 @@ function PrivateTextInput({
 
   const [touched, setTouched] = useState(false)
 
-  const setErrorWrapper = (errorValue) => {
-    if (!!formChangeHandler) {
-      formChangeHandler(value, errorValue)
-    }
-    setError(errorValue)
-  }
-
-  const callErrorHandlers = (value, handlerFn) => {
-    let errorMessage = validate(value)
-    errorMessage = errorMessage.length ? errorMessage : ''
-    handlerFn(errorMessage)
-  }
-
-  const doValidation = (value, isTouched) => {
-    // User hasn't blurred but we still need to inform form
-    // engine if we're in a valid state or not
-    if (!isTouched && !!formChangeHandler) {
-      callErrorHandlers(value, formChangeHandler)
-    } else {
-      // Have blurred
-      callErrorHandlers(value, setErrorWrapper)
-    }
-  }
+  const [doValidation] = useInputValidation({validate, setError, formChangeHandler})
```

Comment:
TextInput is an example of a component who's validation can defer to the "boiler-plate way" so no overrides. It now can just continue to call `doValidation` and all else is taken care of.